### PR TITLE
[14.0][MIG] stock_analytic: Migration to V14.

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

I have migrated the module stock_analytic I also make a little improvement and add analytic groups in the field, in order to show analytic accounts in stock moves only to users that have analytic group.

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

**Comment by rousseldenis** [comment]
There is an issue in another module : https://github.com/OCA/account-analytic/issues/354

**Comment by alan196** [comment]
Now test are OK 👍 

@rousseldenis Could you help me to review?

File: stock_analytic/tests/test_stock_picking.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def _create_picking(
             "location_id": location_id.id,
             "location_dest_id": location_dest_id.id,
             "date": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
-            "date_expected": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+            "date_deadline": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
```
Comment (rousseldenis) [inline stock_analytic/tests/test_stock_picking.py:96]: I think you can do fields.Datetime.now()

File: stock_analytic/models/stock.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    def _prepare_account_move_line(
+        self, qty, cost, credit_account_id, debit_account_id, description
+    ):
+        self.ensure_one()
+        res = super(StockMove, self)._prepare_account_move_line(
+            qty, cost, credit_account_id, debit_account_id, description
+        )
+        for line in res:
+            if (
```
Comment (clementmbr) [inline stock_analytic/models/stock.py:28]: @alan196 @rousseldenis  It's not really a review, but I'm willing to use this module and I don't catch why there is this condition on the product category's *stock valuation account.account.*
Do you know the reason ?
Thank you very much ! :pray:

File: stock_analytic/models/stock.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                line[2]["account_id"]
+                != self.product_id.categ_id.property_stock_valuation_account_id.id
+            ):
```
Comment (clementmbr) [inline stock_analytic/models/stock.py:31]: @alan196 @rousseldenis  It's not really a review, but I'm willing to use this module and I don't catch why there is this condition on the product category's *stock valuation account.account.*
Do you know the reason ?
Thank you very much ! :pray:

**Comment by Jerther** [comment]
@alan196 Are you going to put any more effort into this PR? I see you have two reviews with suggested changes. Have you considered them?

I have a bunch of custom modules that depends on stock_analytic so either this PR gets pushed or else this more recent one does: https://github.com/OCA/account-analytic/pull/391 . I'd rather help with these than create yet another PR with a proposed migration ;) But first I want to make sure one of them is still maintained...

File: stock_analytic/models/stock.py (hunk 3 above)
Comment (Jerther) [reply stock_analytic/models/stock.py:31]: @clementmbr The same condition is in previous versions: 9.0, 10.0, 11.0, (module is not migrated to 12.0) and 13.0, mabe older versions too. I'd say it's fine.

Comment (rvalyi) [reply stock_analytic/models/stock.py:31]: Yes we have checked this with @clementmbr  it is fine indeed.

**Comment by Freni-OSI** [comment]
Hi @alan196, Can you make this PR ready to merge?

**Review by Jerther — APPROVED** [review]
All tests pass. Looks good to me!!!

**Review by patrickrwilson — APPROVED** [review]
Thanks for contributing, LGTM

**Comment by Jerther** [comment]
@dreispt @pedrobaeza If that's ok, would it be possible for you to merge this PR please? I can do without but it'd be really nice to have this module ready in the main repo for our migration to Odoo 14 this weekend.

**Comment by pedrobaeza** [comment]
@rousseldenis do you agree on merging?

**Review by dreispt — APPROVED** [review]

**Comment by Jerther** [comment]
Thank you!!

See you for Odoo 15 ;)

**Pull request merged**